### PR TITLE
[COST-5443] Creating a common filter for filtering by resource_id and instance_name.

### PR DESCRIPTION
**Pull request by bacciotti**
## Jira Ticket

[COST-5443](https://issues.redhat.com/browse/COST-5443)

## Description

This change will drop `instance_name` and `resource_id` filter options for EC2 Compute and insert a new filter named `instance`, which refers to instance_name and resource_id fields.

Additionally, this change will drop `instance_name` and `resource_id` exclude options for EC2 Compute and insert a new exclude named `instance`, which refers to instance_name and resource_id fields.

## Testing

1. Checkout Branch
2. Restart Koku
3. Run tests with `tox -e py311 -- api.report.test.aws.test_views.AWSReportViewTest.test_ec2_compute_instance_exclude_filter` and `tox -e py311 -- api.report.test.aws.test_views.AWSReportViewTest.test_ec2_compute_instance_filter`

## Release Notes
- [ ] Add a common filter for filtering by resource_id and instance_name

```markdown
* [COST-5443](https://issues.redhat.com/browse/COST-5443) Add a common filter for filtering by resource_id and instance_name
```

### CODE REVIEW

File: koku/api/report/aws/serializers.py [hunk 1]
```diff
@@ -215,8 +215,12 @@ class AWSEC2ComputeFilterSerializer(BaseFilterSerializer):
     limit = None
     offset = None
 
-    resource_id = StringOrListField(child=serializers.CharField(), required=False)
-    instance_name = StringOrListField(child=serializers.CharField(), required=False)
+    # Not valid for this endpoint.
+    # instance_name and resource_id is filtered by instance field.
+    instance_name = None
+    resource_id = None
```
Comment (maskarb) [inline koku/api/report/aws/serializers.py:221]: We don't need to do this (these can be removed). `instance_name` and `resource_id` are not defined in the `BaseFilterSerializer` class. We only need to set fields to `None` when we are overriding the base class.

File: koku/api/report/aws/serializers.py [hunk 2]
```diff
@@ -243,12 +247,16 @@ class AWSEC2ComputeFilterSerializer(BaseFilterSerializer):
 class AWSEC2ExcludeSerializer(BaseExcludeSerializer):
     """Serializer for handling query parameter exclude."""
 
-    _opfields = ("account", "region", "resource_id", "instance_name", "operating_system")
+    _opfields = ("account", "region", "instance", "operating_system")
+
+    # Not valid for this endpoint.
+    # instance_name and resource_id is filtered by instance field.
+    instance_name = None
+    resource_id = None
```
Comment (maskarb) [inline koku/api/report/aws/serializers.py:255]: Same thing here, we can remove this.

**Review by maskarb — COMMENTED** [review]
A couple comments:

File: koku/api/report/aws/provider_map.py [hunk 3]
```diff
@@ -70,6 +70,10 @@ def __init__(self, provider, report_type, schema_name, cost_type, markup_cost="m
                     "org_unit_single_level": {"field": "organizational_unit__org_unit_id", "operation": "icontains"},
                     "instance_type": {"field": "instance_type", "operation": "icontains"},
                     "operating_system": {"field": "operating_system", "operation": "icontains"},
+                    "instance": [
+                        {"field": "instance_name", "operation": "icontains", "composition_key": "instance_filter"},
+                        {"field": "resource_id", "operation": "icontains", "composition_key": "instance_filter"},
+                    ],
                     "instance_name": {"field": "instance_name", "operation": "icontains"},
```
Comment (lcouzens) [inline koku/api/report/aws/provider_map.py:77]: should this be removed?

File: koku/api/report/aws/provider_map.py [hunk 4]
```diff
@@ -70,6 +70,10 @@ def __init__(self, provider, report_type, schema_name, cost_type, markup_cost="m
...
                     "resource_id": {"field": "resource_id", "operation": "icontains"},
```
Comment (lcouzens) [inline koku/api/report/aws/provider_map.py:78]: and this?

File: koku/api/report/aws/serializers.py (hunk 1 above)
Comment (lcouzens) [inline koku/api/report/aws/serializers.py:221]: is it necessary to have these None's in here?

File: koku/api/report/aws/serializers.py (hunk 2 above)
Comment (lcouzens) [inline koku/api/report/aws/serializers.py:255]: same here?

**Review by lcouzens — COMMENTED** [review]
couple of questions

**Review by lcouzens — APPROVED** [review]

**Pull request merged**